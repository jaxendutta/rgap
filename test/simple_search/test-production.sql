-- Sample query for sp_search_recipients (user id = 42)
SELECT 
  r.recipient_id,
  r.legal_name,
  i.name AS research_organization_name,
  i.city,
  i.province,
  i.country,
  COUNT(DISTINCT rg.grant_id) AS grant_count,
  COALESCE(SUM(rg.agreement_value), 0) AS total_funding,
  MAX(rg.agreement_start_date) AS latest_grant_date,
  IF(
    1 IS NOT NULL,
    EXISTS(
      SELECT 1 
      FROM BookmarkedRecipients br 
      WHERE br.user_id = 1 AND br.recipient_id = r.recipient_id
    ),
    FALSE
  ) AS is_bookmarked
FROM Recipient r
JOIN Institute i ON r.institute_id = i.institute_id
LEFT JOIN ResearchGrant rg ON r.recipient_id = rg.recipient_id
WHERE r.legal_name LIKE CONCAT('%', 'university', '%')
   OR i.name LIKE CONCAT('%', 'university', '%')
GROUP BY r.recipient_id
ORDER BY total_funding DESC
LIMIT 10 OFFSET 0;
