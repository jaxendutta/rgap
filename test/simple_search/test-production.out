+--------------+-----------------------------------------------+--------------------------------+-----------+----------+---------+-------------+---------------+-------------------+---------------+
| recipient_id | legal_name                                    | research_organization_name     | city      | province | country | grant_count | total_funding | latest_grant_date | is_bookmarked |
+--------------+-----------------------------------------------+--------------------------------+-----------+----------+---------+-------------+---------------+-------------------+---------------+
|        48121 | University of Toronto                         | University of Toronto          | Toronto   | ON       | CA      |          50 |  563224716.00 | 2024-06-01        |             1 |
|        48091 | McGill University                             | McGill University              | Montreal  | QC       | CA      |          44 |  352018809.00 | 2024-04-01        |             0 |
|         3718 | The University of British Columbia            | University of British Columbia | Vancouver | BC       | CA      |          61 |  245939355.00 | 2024-06-01        |             1 |
|         3747 | University of Calgary                         | University of Calgary          | Calgary   | AB       | CA      |          35 |  241187491.00 | 2024-06-01        |             0 |
|         3727 | University of Ottawa                          | University of Ottawa           | Ottawa    | ON       | CA      |          38 |  212661333.00 | 2024-06-01        |             0 |
|         3737 | Dalhousie University                          | Dalhousie University           | Truro     | NS       | CA      |          26 |  199004611.00 | 2023-09-01        |             0 |
|         3776 | University of Alberta                         | University of Alberta          | Edmonton  | AB       | CA      |          41 |  170048085.00 | 2024-04-01        |             0 |
|        10528 | Concordia University                          | Concordia University           | Montreal  | QC       | CA      |          27 |  164652502.00 | 2023-04-01        |             0 |
|         3753 | York University                               | York University                | Toronto   | ON       | CA      |          33 |  156081305.00 | 2023-04-01        |             0 |
|        48990 | University of Toronto - University of Toronto | University of Toronto          | Toronto   | ON       | CA      |           4 |  144470973.00 | 2019-04-01        |             0 |
+--------------+-----------------------------------------------+--------------------------------+-----------+----------+---------+-------------+---------------+-------------------+---------------+
10 rows in set (0.95 sec)
